src/librustc_trans/adt.rs

```diff
@@ -162,284 +161,208 @@ pub fn represent_type<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
-            let hint = *cx.tcx().lookup_repr_hints(def.did).get(0)
-                .unwrap_or(&attr::ReprAny);
-
-            if cases.is_empty() {
-                // Uninhabitable; represent as unit
-                // (Typechecking will reject discriminant-sizing attrs.)
-                assert_eq!(hint, attr::ReprAny);
-                return Univariant(mk_struct(cx, &[], false, t));
-            }
+fn integer_to_int_type(from: layout::Integer, signed: bool)->IntType {
+    use rustc::ty::layout::Integer::*;
+    use syntax::attr::IntType::*;
+    use syntax::ast::{UintTy, IntTy};
+    match (from, signed) {
+        (I1, false) => UnsignedInt(UintTy::U8),
+        (I8, false) => UnsignedInt(UintTy::U8),
+        (I16, false) => UnsignedInt(UintTy::U16),
+        (I32, false) => UnsignedInt(UintTy::U32),
+        (I64, false) => UnsignedInt(UintTy::U64),
+        (I1, true) => SignedInt(IntTy::I8),
+        (I8, true) => SignedInt(IntTy::I8),
+        (I16, true) => SignedInt(IntTy::I16),
+        (I32, true) => SignedInt(IntTy::I32),
+        (I64, true) => SignedInt(IntTy::I64),
+    }
+}
 
-            if cases.iter().all(|c| c.tys.is_empty()) {
-                // All bodies empty -> intlike
-                let discrs: Vec<_> = cases.iter().map(|c| Disr::from(c.discr)).collect();
-                let bounds = IntBounds {
-                    ulo: discrs.iter().min().unwrap().0,
-                    uhi: discrs.iter().max().unwrap().0,
-                    slo: discrs.iter().map(|n| n.0 as i64).min().unwrap(),
-                    shi: discrs.iter().map(|n| n.0 as i64).max().unwrap()
-                };
-                return mk_cenum(cx, hint, &bounds);
-            }
+fn mk_mach_int_from_layout<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, tm: layout::Integer, signed: bool)->Ty<'tcx> {
+    use rustc::ty::layout::Integer::*;
+    match (tm, signed) {
+        (I1, false) => tcx.types.u8,
+        (I8, false) => tcx.types.u8,
+        (I16, false) => tcx.types.u16,
+        (I32, false) => tcx.types.u32,
+        (I64, false) => tcx.types.u64,
+        (I1, true) => tcx.types.i8,
+        (I8, true) => tcx.types.i8,
+        (I16, true) => tcx.types.i16,
+        (I32, true) => tcx.types.i32,
+        (I64, true) => tcx.types.i64,
+    }
+}
 
-            // Since there's at least one
-            // non-empty body, explicit discriminants should have
-            // been rejected by a checker before this point.
-            if !cases.iter().enumerate().all(|(i,c)| c.discr == Disr::from(i)) {
-                bug!("non-C-like enum {} with specified discriminants",
-                     cx.tcx().item_path_str(def.did));
+fn convert_layout_struct<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
+                               from: &layout::Struct,
+                               size: layout::Size,
+                               variant: Option<ty::VariantDef<'tcx>>,
+                               substs: &Substs<'tcx>)->Struct<'tcx> {
```

Comment:
The arguments should be indented 4 more to the right (to align with `tcx: ...`) and there should be spaces around `->`.